### ISSUE TITLE
What's the right way to generate a pernosco trace that can be uploaded later?

### ISSUE THREAD

**Issue by KJTsanaktsidis** [Pernosco/pernosco#55]
I'm trying to use rr to find and fix flaky tests in the Ruby language test suite. I was hoping to set up a Jenkins server that did something along these lines:

* Reguarly run the Ruby test suite under RR (one rr invocation per test file)
* When a test fails, upload a packed rr trace as an artifact in Jenkins
* I would look at the test failure, and if it was interesting,
* Then I could download the packed trace and upload it to pernosco.

Unfortunately, I'm having a lot of trouble getting the sources to appear in pernosco. What I'm doing is:

* On the Jenkins machine, run the following after `rr` has exited with status 1
  * `rr pack $trace_dir`
  * `pernosco-submit analyze-build --build-dir . --allow-source /usr/include --allow-source /usr/lib --allow-source /usr/local/include --allow-source /usr/local/lib --copy-sources .. $trace_dir miniruby ruby **/*.so` (this seems to create a directory `extra_rr_trace_files` in the trace)
  * `tar -cxf rr_trace.tar.gz $trace_dr`
* Downloading the `rr_trace.tar.gz` file onto my machine (which won't have the sources!) and extracting it
* Running `pernosco-submit upload --title 'foo' --no-local-sources rr_trace`

Pernosco doesn't seem to show any sources though for this trace. It knows the file names/line numbers from the debuginfo, but no source text:

![image](https://github.com/user-attachments/assets/c2cc9a7c-7d15-4a59-952d-faa9d6bfd8b5)

You can see the `rr_trace.tar.gz` file on my jenkins server here: https://ruby-rr-ci.kjtsanaktsidis.id.au/jenkins/job/ruby_ci-rr-asan/206/testReport/test-all/TestYJIT/TestYJIT_test_code_gc_partial_last_page/attachments/rr_trace.tar.gz

I know that `pernosco-submit` has a `--dry-run` option, but I didn't think this would be suitable for this use-case because it signs the trace with my pernosco credentials, right? I guess I would want CI to produce an "unsigned" archive for this and allow people interested in the trace to sign & upload it with their own credentials.

Can you tell me how exactly I'm holding this wrong? Thanks!

**Comment by khuey** [comment]
Yeah seems like we need a "package everything but don't sign it" option. The on-premises version has a "package-build" command so I think I'll do that.

**Comment by KJTsanaktsidis** [comment]
That sounds like exactly what I need 👍 in the meanwhile I guess I’m the only one actually using this system so I’ll just upload my signed traces 🤷 (as long as this isn’t an abuse vector for you guys?)

**Comment by khuey** [comment]
Yeah that's fine (sorry I didn't see your reply earlier).

**Comment by khuey** [comment]
This is supported now.

https://github.com/Pernosco/pernosco-submit/blob/master/README.md#two-step-upload

If you have any issues please file them against the pernosco-submit repo.
